platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,493 @@
+   stored here.
+   When a new candidate roster is received, it will be stored in this map, with the `RosterState`'s current candidate
+   roster hash updated.
+
+### On Disk
+
+There will continue to be components used in creating the Roster - such as the private EC (elliptic curve) key, the RSA
+private key and the signing X509Certificate certificate - that will be stored on disk.
+However, it is at the discretion of the Services team to manage the lifecycle of these files, and not the platform.
+
+There will not be any other separate artifacts from this proposal stored directly on disk.
+
+## Services changes (Inversion of control)
+
+The current startup procedure will be altered as follows
+
+- The `config.txt` file will no longer be used by the platform for storing the address book in existing networks. It
+  will no longer be used by the platform code
+- Designating `config.txt` introduces inversion of control. It will be at the exclusive prerogative of Services (the
+  App) going forward. Platform will no longer be responsible for it, and all the platform code that builds an
+  AddressBook from `config.txt` will be removed or refactored and moved into the Services codebase.
+
+### DevOps Workflow changes
+
+The current network Transplant procedure is manual.
+DevOps get given a State and `config.txt` file on disk. This is then followed by a software-only upgrade to adopt
+the `config.txt` file using that state. This will change to the use of a Genesis Roster.
+
+#### Genesis Roster
+
+A `Genesis Roster` is an optional Roster that DevOps may provide to either start a Genesis Network process or Transplant
+an existing network. It is essential to distinguish the Genesis Roster from the Candidate or Active rosters.
+The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network. However, its structure is
+exactly the same as the Active or Candidate Rosters.
+The equivalent of the Genesis Roster in the current network is the `config.txt` file.
+DevOps and Services may choose to create the Genesis Roster from the `config.txt` file or some other mechanism.
+
+A new method will be added to the `PlatformBuilder` that will be used by the App to set the Genesis Roster.
+In most cases, this Genesis Roster will be null.
+
+```java
+//in PlatformBuilder
+
+/*
+  Set the Genesis Roster for the network.
+  <p>
+  This method is used to set the optional Genesis Roster for the network.
+  The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network.
+  The Genesis Roster, if set, is immutable and will be used to start a Genesis network Process or Network Transplant process.
+  <p>
+  @param genesisRoster the Genesis Roster for the network.
+ */
+void withGenesisRoster(@NonNull final Roster genesisRoster);
+```
+
+The logic to determine whether to invoke this method could be determined by a config flag,
```

Comment:
I don't think we need to mention the use of a config flag here.   That would be between Services and DevOps.

Reply:
I want to mention this will require some new logic without the details. I've rephrased it now.